best_practices.md

```diff
@@ -0,0 +1,195 @@
+more reliable and faster than developing your own solution.
+Therefore it is highly recommended that you make use of existing libraries
+and contribute to existing software instead of developing your own software
+from scratch.
+
+Programming languages come with a large collection of built-in functions and
+libraries, called a standard library.
+It is recommended that you make use of functions from the standard library.
+An example of such a library is the
+[Python standard library](https://docs.python.org/3/library/).
+
+Climate data is typically stored in NetCDF format using the CF-Conventions.
+Usually climate data is larger than the amount of memory available in a typical
+computer.
+A programming language that is suitable for analyzing climate data has libraries
+that support
+- reading and writing NetCDF files or an equivalent storage format supporting the CF-Conventions
+- understand the CF-Conventions
+- computing statistics on arrays that are larger than memory
+- creating figures
+
+A good example of a programming language that meets all these requirements is Python,
+where the [xarray](http://xarray.pydata.org) or
+[iris](https://scitools-iris.readthedocs.io) libraries facilitate the processing
+of large climate data, while [matplotlib](https://matplotlib.org/) or
+[altair](https://altair-viz.github.io/) can be used for visualization.
+
+## Follow code quality standards
+
+It is recommended that you make use of the
+[style guide](https://the-turing-way.netlify.app/reproducible-research/code-quality.html)
+that is available for the programming language of your choice.
+Examples of style guides:
+- [PEP8](https://www.python.org/dev/peps/pep-0008/) and [PEP257](https://www.python.org/dev/peps/pep-0257/) for Python
+- The [Tidyverse style guide ](https://style.tidyverse.org/)for R
+
+If there is no style guide available for the programming language of your
+choice, you may want to reconsider that choice because the language may not be
+mature enough.
+If there are compelling reasons to choose a language without a style guide,
+it is recommended that you at least
+- organize your code in multiple files of no more than a few hundred lines each
+- organize your code in functions that fit your screen without scrolling (typically no more than 50 lines)
+- use easy to understand variable names
+- insert documentation or comments in your code that explain what is going on
+
+This kind of organization increases readability of your code, thereby reducing
+the risk of mistakes and making your code accessible to others.
+
+Style guides typically come with automatic formatters, that can automatically
+format your code according to the chosen style.
+In addition to automatic formatters, tools for checking adherence to a style
+guide as well as for common programming mistakes are usually also available.
+These tools are called linters.
+It is recommended that you make use of an automatic formatter and linter.
```

Comment:
It comes later in the document ... but mentioning CI here for automation?

Reply:
Thanks, I included some text and links